contracts/main/CurveTricryptoOptimizedWETH.vy

```diff
@@ -1764,6 +1721,40 @@ def price_oracle(k: uint256) -> uint256:
     return price_oracle
 
 
+@external
+@view
+@nonreentrant("lock")
+def xcp_oracle() -> uint256:
+    """
+    @notice Returns the oracle value for xcp.
+    @dev The oracle is an exponential moving average, with a periodicity
+         determined by `self.ma_time`.
+         `TVL` is xcp, calculated as either:
+            1. virtual_price * total_supply, OR
+            2. self.get_xcp(...), OR
+            3. MATH.geometric_mean(xp)
+    @return uint256 Oracle value of xcp.
+    """
+
+    last_prices_timestamp: uint256 = self.last_prices_timestamp
+    cached_xcp_oracle: uint256 = self.cached_xcp_oracle
+
+    if last_prices_timestamp < block.timestamp:
+
+        ma_time: uint256 = self._unpack(self.packed_rebalancing_params)[2]
+        alpha: uint256 = MATH.wad_exp(
+            -convert(
+                (block.timestamp - last_prices_timestamp) * 10**18 / ma_time,
+                int256,
+            )
+        )
+
+        last_xcp: uint256 = self.get_xcp(self.D, self.price_scale_packed)
+        return (last_xcp * (10**18 - alpha) + cached_xcp_oracle * alpha) / 10**18
```

Comment:
No, no, this is wrong!! You do like we are doing with price oracles now:
```
output = self.last_xcp * (1 - alpha) + previous_oracle * alpha
self.last_xcp = self.get_xcp(...)
```

Otherwise your value can jump - same problem we have with an external oracle now. Yes you'd need to save two variables

Reply:
Fixed.